processor/ratelimitprocessor/internal/sharedcomponent/sharedcomponent.go

```diff
@@ -0,0 +1,182 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Package sharedcomponent exposes functionality for components
+// to register against a shared key, such as a configuration object, in order to be reused across signal types.
+// This is particularly useful when the component relies on a shared resource such as os.File or http.Server.
+package sharedcomponent // import "github.com/elastic/opentelemetry-collector-components/processor/ratelimitprocessor/internal/sharedcomponent"
```

Comment:
WDYT about moving this as separate package given we are also using in elasticapmreceiver - https://github.com/elastic/opentelemetry-collector-components/tree/main/receiver/elasticapmreceiver/internal/sharedcomponent ?

Reply:
Thanks for the details, I cant think of a way from top of my head. Its not a blocker per se. I am fine with keeping it as it is.